server.js

```diff
@@ -2,36 +2,84 @@ import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+import netflixData from './data/netflix-titles.json'
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const Movie = mongoose.model('Movie', {
+  title: String,
+  release_year: Number,
+  type: String,
+  rating: String
+})
+
+const seedDatabase = async () => {
+  await Movie.deleteMany()
+  netflixData.forEach((movie) => new Movie(movie).save())
+
+  new Movie({ title: 'Titanic', release_year: 1997, type: 'Movie' }).save()
+  new Movie({ title: 'Sons of Anarchy', release_year: 2008, type: 'TV Show' }).save()
+  new Movie({ title: 'Forrest Gump', release_year: 1994, type: 'Movie' }).save()
+}
+
+seedDatabase()
+
 const port = process.env.PORT || 8080
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable' })
+  }
+})
+
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  // res.send('Hello world')
+  Movie.find().then(movies => {
+    res.json(movies)
+  })
 })
 
+//Find by title: http://localhost:8080/title/Titanic
+app.get('/title/:title', (req, res) => {
+  Movie.findOne({ title: req.params.title }).then(movie => {
+    if (movie) {
+      res.json(movie)
+    } else {
+      res.status(404).json({ error: 'Movie not found!' })
+    }
+  })
+})
```

Comment:
Endpoints' names should be in plural (and then the path param could be in singular): `/titles/:title`.

What happens if two movies has the same name? `findOne` just returns the first match. Consider searching for something unique, like `id` instead.